client/landing/stepper/declarative-flow/internals/components/step-route/hooks/use-step-route-tracking/index.tsx

```diff
@@ -25,27 +30,44 @@ const useIsRequestingSelectedSite = ( siteSlugOrId: string | number, site: SiteD
 };
 
 interface Props {
-	flowName: string;
+	flow: Flow;
 	stepSlug: string;
 	// If true, the tracking event will not be recorded
 	skipTracking: boolean;
-	flowVariantSlug?: string;
 }
 
 /**
  * Hook to track the step route in the declarative flow.
  */
-export const useStepRouteTracking = ( {
-	flowName,
-	stepSlug,
-	skipTracking,
-	flowVariantSlug,
-}: Props ) => {
+export const useStepRouteTracking = ( { flow, stepSlug, skipTracking }: Props ) => {
 	const intent = useIntent();
 	const design = useSelectedDesign();
 	const { site, siteSlugOrId } = useSiteData();
 	const isRequestingSelectedSite = useIsRequestingSelectedSite( siteSlugOrId, site );
 	const hasRequestedSelectedSite = siteSlugOrId ? !! site && ! isRequestingSelectedSite : true;
+	const signupStepStartProps = useSnakeCasedKeys( {
+		input: flow.useTracksEventProps?.()?.[ STEPPER_TRACKS_EVENT_SIGNUP_STEP_START ],
+	} );
+
+	const recordStepStart = useCallback(
+		( step: string ) => {
+			recordTracksEvent( STEPPER_TRACKS_EVENT_SIGNUP_STEP_START, {
+				flow: flow.name,
+				step: kebabCase( step ),
+				device: resolveDeviceTypeByViewPort(),
+				intent,
+				is_in_hosting_flow: isAnyHostingFlow( flow.name ),
+				...( design && { assembler_source: getAssemblerSource( design ) } ),
+				...( flow.variantSlug && { flow_variant: flow.variantSlug } ),
+				...signupStepStartProps,
+			} );
+		},
+
+		// We leave out intent and design from the dependency list, due to the ONBOARD_STORE being reset in the exit flow.
+		// The store reset causes these values to become empty, and may trigger this event again.
```

Comment:
Hopefully, we'll move design, intent, and `is_in_hosting_flow` out of here into the flows.

Reply:
Also hopefully by the end of these refactors we'll have a generalised utility that will record these values by default (once all the framework events are handled similarly)